### ISSUE TITLE
fit_generator fills up disk in no time

### ISSUE THREAD

**Issue by eaigner** [keras-team/keras#8655]
I'm generating image data on the fly with `fit_generator`. After just a few minutes, the generator fills up the disk with Gigabytes of data.

1) Why does an on-they-fly generator use disk space at all?
2) Why does it use SO MUCH data? The images are `416 * 416 * 3 * 800` bytes which should only equate to ~415 MB / epoch

**Comment by Dref360** [comment]
Hi,
Quick questions : 
1- Are you on windows?
2- Could you post your generator?
Thanks,

**Comment by jermenkoo** [comment]
I use a generator (using Python 3.6.3 w/ Keras 2.0.8 and TensorFlow 1.3.0) and my generator works just fine.

The code I have is pretty similar to this one:

```
def sintel_generator(batch_size, val=False, test=False):
    while 1:
        X = np.zeros(shape=(batch_size, 256, 512, 3 * 2), dtype="uint8")
        y = np.zeros(shape=(batch_size, 256, 512, 3), dtype="uint8")

        batch_num = 0
        while batch_num < batch_size:
            # select an image from a random folder

            # read images
            X[batch_num, :, :, :3] = imread(os.path.join(...))[:, :, :])
            X[batch_num, :, :, 3:] = imread(os.path.join(...))[:, :, :])
            y[batch_num, :, :] = imread(os.path.join(...))[:, :, :])

            yield X, y
            batch_num += 1
```

It works just fine - mind posting code for your generator? Are you doing any data augmentation?

**Comment by eaigner** [comment]
I think its a swapping issue. As soon as i did `del big_obj` in my generator the disk didn't fill up anymore. The strange thing is I would have more than enough memory to support keeping the objects in memory. Also the Python process memory was fixed at around 1.4GB (32GB total), which seems strange.

**Comment by Dref360** [comment]
Try updating to master, we've just fixed a bug about a similar issue.
